common/src/main/java/org/conscrypt/HpkeSpi.java

```diff
@@ -0,0 +1,88 @@
+package org.conscrypt;
+
+import java.security.InvalidKeyException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import javax.crypto.BadPaddingException;
+
+/**
+ * SPI for HPKE clients to communicate with implementations.  The client API can use any
+ * implementation which implements this interface, by duck-typing if necessary.
+ */
+public interface HpkeSpi {
+
+  /**
+   * Initialises an HPKE sender SPI.
+   *
+   * @param mode HPKE mode to use
+   * @param publicKey public key of the recipient
+   * @param info application-supplied information, may be null or empty
+   * @param sKe optional random seed, should be null for all uses except for validation against
+   *            known test vectors
+   * @throws InvalidKeyException if publicKey is null or an unsupported key format
+   * @throws UnsupportedOperationException if mode is not a supported HPKE mode
+   * @throws IllegalStateException if this SPI has already been initialised
+   */
+  void engineInitSender(int mode, PublicKey publicKey, byte[] info, byte[] sKe)
```

Comment:
Same comment regarding "mode". I think it would be better to remove the parameter. If we ever support all other modes, those modes could be includes as additional methods instead. The other modes require additional parameters.